pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -654,32 +708,77 @@ func (c *InstallerController) newNodeStateForInstallInProgress(ctx context.Conte
 			return currNodeState, false, fmt.Sprintf("waiting for static pod of revision %d, found %s", currNodeState.TargetRevision, currentRevision), nil
 		}
 
+		startupMonitor, err := c.startupMonitorEnabled()
+		if err != nil {
+			return currNodeState, false, "", err
+		}
+
 		switch state {
 		case staticPodStateFailed:
+			ns, name := c.targetNamespace, mirrorPodNameForNode(c.staticPodName, currNodeState.NodeName)
+			if len(staticPodErrors) == 0 {
+				staticPodErrors = append(staticPodErrors, fmt.Sprintf("no detailed termination message, see `oc get -oyaml -n %q pods %q`", ns, name))
+			}
+
 			ret := currNodeState.DeepCopy()
 			ret.LastFailedRevisionErrors = staticPodErrors
-			ret.TargetRevision = 0 // stop installer retries
-			ret.LastFailedRevision = currNodeState.TargetRevision
-			ret.LastFailedTime = &metav1.Time{Time: ts}
-			ret.LastFailedCount++
-			ret.LastFailedReason = NodeStatusOperandFailedReason
 
-			ns, name := c.targetNamespace, mirrorPodNameForNode(c.staticPodName, currNodeState.NodeName)
-			if len(ret.LastFailedRevisionErrors) == 0 {
-				ret.LastFailedRevisionErrors = []string{fmt.Sprintf("no detailed termination message, see `oc get -oyaml -n %q pods %q`", ns, name)}
+			if startupMonitor {
+				// this will be overridden again when the fallback pod is observed
+				ret.LastFailedRevisionErrors = append(ret.LastFailedRevisionErrors, "falling back to last-known-good revision")
+			} else {
+				ret.TargetRevision = 0 // stop installer retries, but keep running in startup-monitor case where we leave this to the startup-monitor to do
+				ret.LastFailedRevision = ret.TargetRevision
```

Comment:
remove, as it is set in the next line (733)

Reply:
fixed